kapitan/remoteinventory/fetch.py

```diff
@@ -0,0 +1,201 @@
+    logger.info("Inventory {} : fetching now".format(source))
+    try:
+        Repo.clone_from(source, os.path.join(save_dir, os.path.basename(source)))
+        logger.info("Inventory {} : successfully fetched".format(source))
+    except GitCommandError as e:
+        logger.error(e)
+        raise GitFetchingError("Inventory {} : fetching unsuccessful{}".format(source, e.stderr))
+
+
+def fetch_http_inventories(inv_mapping, inventory_path, save_dir):
+    """Fetches inventory items over http[s] and saves it to save_dir which is later recursively
+       copied to the output_path stored in inv_mapping
+
+       :param inv_mapping: tupple containing source and list of invetory variables
+       :param inventory_path: default or user defined inventory path
+       :param save_dir: directory to same the fetched inventory items into
+
+       :returns: None
+    """
+    source, invs = inv_mapping
+    content_type = fetch_http_source(source, save_dir)
+    path_hash = hashlib.sha256(os.path.dirname(source).encode()).hexdigest()[:8]
+    copy_src_path = os.path.join(save_dir, path_hash + os.path.basename(source))
+    for inv in invs:
+        output_path = os.path.join(inventory_path, inv["output_path"])
+        if inv.get("unpack", False):
+            # ensure that the directory we are extracting to exists
+            os.makedirs(output_path, exist_ok=True)
+            is_unpacked = unpack_downloaded_file(copy_src_path, output_path, content_type)
+            if is_unpacked:
+                logger.info("Inventory {} : extracted to {}".format(source, output_path))
+            else:
+                logger.info(
+                    "Inventory {} : Content-Type {} is not supported for unpack. Ignoring save".format(
+                        source, content_type
+                    )
+                )
+        else:
+            # we are downloading a single inventory item
+            parent_dir = os.path.dirname(output_path)
+            if parent_dir != "":
+                os.makedirs(parent_dir, exist_ok=True)
+            copyfile(copy_src_path, output_path)
+            logger.info("Inventory item {} : saved to {}".format(source, output_path))
+
+
+def fetch_http_source(source, save_dir):
+    """downloads a http[s] file from source and saves into save_dir
+       :returns: content_type
+    """
+    logger.info("Dependency {} : fetching now".format(source))
+    content, content_type = make_request(source)
+    logger.info("Dependency {} : successfully fetched".format(source))
+    if content is not None:
+        basename = os.path.basename(source)
+        # to avoid collisions between basename(source)
+        path_hash = hashlib.sha256(os.path.dirname(source).encode()).hexdigest()[:8]
+        full_save_path = os.path.join(save_dir, path_hash + basename)
+        with open(full_save_path, "wb") as f:
+            f.write(content)
+        return content_type
+    else:
```

Comment:
You can remove this "else:" line here. 
If the `if content is not None:` line isn't true it will just get to the last two lines even without the else: condition.